solr/core/src/java/org/apache/solr/handler/admin/api/ReplaceNodeAPI.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.solr.jersey.PermissionName;
+import org.apache.solr.jersey.SolrJerseyResponse;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.response.SolrQueryResponse;
+
+/**
+ * V2 API for recreating replicas in one node (the source) on another node(s) (the target).
+ *
+ * <p>This API is analogous to the v1 /admin/collections?action=REPLACENODE command.
+ */
+@Path("cluster/nodes/{sourceNodeName}/commands/replace/")
+public class ReplaceNodeAPI extends AdminAPIBase {
+
+  @Inject
+  public ReplaceNodeAPI(
+      CoreContainer coreContainer,
+      SolrQueryRequest solrQueryRequest,
+      SolrQueryResponse solrQueryResponse) {
+    super(coreContainer, solrQueryRequest, solrQueryResponse);
+  }
+
+  @POST
+  @Produces({"application/json", "application/xml", BINARY_CONTENT_TYPE_V2})
+  @PermissionName(COLL_EDIT_PERM)
+  public SolrJerseyResponse replaceNode(
+      @Parameter(description = "The name of the node to be replaced.", required = true)
+          @PathParam("sourceNodeName")
+          String nodeName,
+      @RequestBody(description = "Contains user provided parameters", required = true)
+          ReplaceNodeRequestBody requestBody)
+      throws Exception {
+    final SolrJerseyResponse response = instantiateJerseyResponse(SolrJerseyResponse.class);
+    final CoreContainer coreContainer = fetchAndValidateZooKeeperAwareCoreContainer();
+    /** TODO Record node for log and tracing */
+    final ZkNodeProps remoteMessage = createRemoteMessage(nodeName, requestBody);
+    final SolrResponse remoteResponse =
+        CollectionsHandler.submitCollectionApiCommand(
+            coreContainer,
+            coreContainer.getDistributedCollectionCommandRunner(),
+            remoteMessage,
+            CollectionParams.CollectionAction.REPLACENODE,
+            DEFAULT_COLLECTION_OP_TIMEOUT);
+    if (remoteResponse.getException() != null) {
+      throw remoteResponse.getException();
+    }
+
+    disableResponseCaching();
+    return response;
+  }
+
+  public ZkNodeProps createRemoteMessage(String nodeName, ReplaceNodeRequestBody requestBody) {
+    final Map<String, Object> remoteMessage = new HashMap<>();
+    remoteMessage.put(SOURCE_NODE, nodeName);
+    remoteMessage.put(TARGET_NODE, requestBody.targetNodeName);
+    remoteMessage.put(WAIT_FOR_FINAL_STATE, requestBody.waitForFinalState);
+    remoteMessage.put(QUEUE_OPERATION, CollectionAction.REPLACENODE.toLower());
+
+    return new ZkNodeProps(remoteMessage);
+  }
+
+  public static class ReplaceNodeRequestBody implements JacksonReflectMapWriter {
+
+    public ReplaceNodeRequestBody() {}
+
+    public ReplaceNodeRequestBody(String targetNodeName, Boolean waitForFinalState) {
+      this.targetNodeName = targetNodeName;
+      this.waitForFinalState = waitForFinalState;
+    }
+
+    @Schema(description = "The name of the targetNode.")
+    @JsonProperty("targetNodeName")
+    public String targetNodeName;
+
+    @JsonProperty("waitForFinalState")
```

Comment:
[0] The `@Schema` annotations you see on this code end up used as descriptions in the OpenAPI spec that Solr generates as part of its build.

If you're not familiar with OpenAPI, it's really awesome! It provides tooling you can use to generate a "spec" - a machine-readable description of your API.  And then other tooling can take that spec and do all sorts of things with it, like generating clients for various languages.

Right now, Solr's set up to generate a spec (run `./gradlew resolve` and take a look at `solr/core/build/generated/openapi/openapi.json`).  We haven't started using the spec yet, but it's on my roadmap to set us up to generate Solr clients for Python, golang, etc.

Anyway, this is all to say: these `@Schema` annotations don't seem like they do much now, but they'll be super helpful to have in place once we start using the spec.

You don't need to have them everywhere if you're not sure what to say about certain parameters.  But if you want to include them, you can always copy the snippet in the ref guide that describes the parameter.